pkg/stanza/fileconsumer/file.go

```diff
@@ -73,7 +77,7 @@ func (m *Manager) Stop() error {
 		m.cancel = nil
 	}
 	m.wg.Wait()
-	m.tracker.ClosePreviousFiles()
+	m.openFiles.Add(context.TODO(), int64(0-m.tracker.ClosePreviousFiles()))
```

Comment:
Are you willing to look into what it takes to pass the context through?

Reply:
If I'm not mistaken we could pass down the `ctx` from the `Shutdown` of the stanza receiver when we stop the internal  pipeline at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d78d7bb37d0c84da04ac8a83356eb669ecd75a95/pkg/stanza/adapter/receiver.go#L142. 

Then if we pass the `ctx` on every single operator's `Stop()` we can make it available to the `Stop()` of the `Manager`/fileconsumer at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d78d7bb37d0c84da04ac8a83356eb669ecd75a95/pkg/stanza/operator/input/file/input.go#L36 when we stop `file_input` operator. 

This would be doable I think but would be a breaking change for the [`Pipeline`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5133f4ccd69fa40d016c5b7f2198fb6ac61007b4/pkg/stanza/pipeline/pipeline.go#L13) interface and the [`Operator`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d78d7bb37d0c84da04ac8a83356eb669ecd75a95/pkg/stanza/operator/operator.go#L24) interface (plus the `Manager`).
Hence, I'm not sure if that's worth doing it unless we find value in doing so anyways and not just for the meter's need?